src/Components/Components/src/Rendering/HtmlRenderer.cs

```diff
@@ -47,20 +47,20 @@ protected override Task UpdateDisplayAsync(in RenderBatch renderBatch)
         /// <param name="componentType">The type of the <see cref="IComponent"/>.</param>
         /// <param name="initialParameters">A <see cref="ParameterCollection"/> with the initial parameters to render the component.</param>
         /// <returns>A <see cref="Task"/> that on completion returns a sequence of <see cref="string"/> fragments that represent the HTML text of the component.</returns>
-        public async Task<IEnumerable<string>> RenderComponentAsync(Type componentType, ParameterCollection initialParameters)
+        public async Task<RenderedHtmlResult> RenderComponentAsync(Type componentType, ParameterCollection initialParameters)
         {
-            var frames = await CreateInitialRenderAsync(componentType, initialParameters);
+            var (componentId, frames) = await CreateInitialRenderAsync(componentType, initialParameters);
 
             if (frames.Count == 0)
             {
-                return Array.Empty<string>();
+                return new RenderedHtmlResult(-1, Array.Empty<string>());
```

Comment:
What's the case where this occurs? seems worth a comment

Reply:
You are right. This was an oversight on my part. This happens when the root component produces nothing. We simply need to return the root component along with the empty string array and be done with it.